tests/moead_test.cpp

```diff
@@ -776,36 +654,16 @@ TEST_CASE("MOEADDIRICHLETMAF4Test", "[MOEADTest]")
       upperBound // Upper bound.
     );
 
-  typedef decltype(MAF_FOUR.objectiveF1) ObjectiveTypeA;
-  typedef decltype(MAF_FOUR.objectiveF2) ObjectiveTypeB;
-  typedef decltype(MAF_FOUR.objectiveF3) ObjectiveTypeC;
+  typedef decltype(ZDT_THREE.objectiveF1) ObjectiveTypeA;
+  typedef decltype(ZDT_THREE.objectiveF2) ObjectiveTypeB;
 
-  std::tuple<ObjectiveTypeA, ObjectiveTypeB, ObjectiveTypeC> objectives = 
-      MAF_FOUR.GetObjectives();
+  TestType coords = ZDT_THREE.GetInitialPoint();
+  std::tuple<ObjectiveTypeA, ObjectiveTypeB> objectives = ZDT_THREE.GetObjectives();
 
-  bool success = false;
-  arma::mat coords = MAF_FOUR.GetInitialPoint();
   opt.Optimize(objectives, coords);
-  arma::cube paretoSet = opt.ParetoSet();
-  for (size_t i = 0; i < paretoSet.n_slices; i++)
-  {
-    arma::mat solution = paretoSet.slice(i);
-    bool allInRange = true;
-    for (size_t j = 2; j < MAF_FOUR.GetNumVariables(); j++)
-    {
-    	double val = arma::as_scalar(solution(j));
-     if (!IsInBounds<double>(val, expectedLowerBound, expectedUpperBound, 0.2))
-      {
-        allInRange = false;
-        break;
-      }
-    }
-    if(allInRange)
-    {
-      success = true;
-      break;
-    }
-  }
 
-  REQUIRE(success == true);
-}
\ No newline at end of file
+  const coot::Cube<ElemType>& finalPopulation = opt.ParetoSet();
+  REQUIRE(VariableBoundsCheck(finalPopulation));
+}
+
+#endif
```

Comment:
Correct me if I'm wrong, but it looks like a few tests were removed for MOEAD---should we add them back?